### PR TITLE
feat: sorting event emitter

### PR DESCRIPTION
**Pull request by pr3tori4n**
**Related Issue:** #278

## Summary

add a custom event that fires whenever the list sorting changes

### CODE REVIEW

File: src/components/calcite-value-list/calcite-value-list.tsx [hunk 1]
```diff
@@ -146,16 +148,18 @@ export class CalciteValueList {
...
       this.sortables.push(
         Sortable.create(sortGroup, {
           group: this.guid,
           handle: `.${CSS.handle}`,
-          draggable: "calcite-value-list-item"
+          draggable: "calcite-value-list-item",
+          onUpdate: () => {
+            this.items = Array.from(this.el.querySelectorAll("calcite-value-list-item"));
+            const values = this.items.map((item) => item.value);
+            this.calciteValueListOrderChange.emit(values);
```
Comment (jcfranco) [inline src/components/calcite-value-list/calcite-value-list.tsx]: Is this payload needed? Users can query for the items if needed.

Comment (pr3tori4n) [reply src/components/calcite-value-list/calcite-value-list.tsx]: @dhrumil83 can you weigh in on if you would need this or find this useful?

Comment (dhrumil83) [reply src/components/calcite-value-list/calcite-value-list.tsx]: I just need what moved and where via drag and drop so I can update the array. 
If I get the all values, I can check what changed myself and update.

Comment (pr3tori4n) [reply src/components/calcite-value-list/calcite-value-list.tsx]: ok sounds useful. keeping.

**Comment by pr3tori4n** [comment]
@jcfranco I've responded to your comments, please review.

File: src/components/calcite-value-list/calcite-value-list.tsx [hunk 2]
```diff
@@ -101,6 +101,8 @@ export class CalciteValueList {
 
   @Event() calciteValueListSelectionChange: EventEmitter;
```
Comment (jcfranco) [inline src/components/calcite-value-list/calcite-value-list.tsx]: Can you add some JSDoc to public events?

**Pull request merged**